electrum/transaction.py

```diff
@@ -447,6 +449,22 @@ def match_script_against_template(script, template) -> bool:
             return False
     return True
 
+def get_script_type_from_output_script(_bytes: bytes) -> Optional[str]:
+    if _bytes is None:
+        return None
+    try:
+        decoded = [x for x in script_GetOp(_bytes)]
+    except MalformedBitcoinScript:
+        return None
+    if match_script_against_template(decoded, SCRIPTPUBKEY_TEMPLATE_P2PKH):
+        return 'p2pkh'
+    if match_script_against_template(decoded, SCRIPTPUBKEY_TEMPLATE_P2SH):
+        return 'p2sh'
+    if match_script_against_template(decoded, SCRIPTPUBKEY_TEMPLATE_P2WPKH):
+        return 'p2wpkh'
+    if match_script_against_template(decoded, SCRIPTPUBKEY_TEMPLATE_P2WSH):
+        return 'p2wsh'
```

Comment:
Note that in many other places in `transaction.py`, a `script_type` of "p2sh" means p2sh-multisig, while a "p2wsh" means p2wsh-multisig.
I guess something will have to be done about that sooner or later anyway... Maybe it's ok to ignore that for now.
However, this is something to keep in mind if you are planning on comparing these script_type strings to determine if certain inputs use similar scripts.